src/pages/PremiumVideoCall/state/usePasscodeAuth/usePasscodeAuth.ts

```diff
@@ -16,41 +17,29 @@ export function getPasscode() {
 }
 
 export function fetchToken(
+    instantVisioToken: string,
     name: string,
     room: string,
     passcode: string,
     create_room = true
 ) {
-    return fetch(`/token`, {
-        method: 'POST',
-        headers: {
-            'content-type': 'application/json',
-        },
-        body: JSON.stringify({
-            user_identity: name,
-            room_name: room,
-            passcode,
-            create_room,
-        }),
-    })
+    const api = new Api(instantVisioToken)
+    return api
+        .createRoom()
+        .then((response) => {
+            return response.roomId
+        })
+        .then((roomId) => {
+            return api.joinRoom(roomId)
+        })
+        .then((response) => {
+            const { jwtAccessToken } = response
+            return jwtAccessToken
+        })
 }
 
 export function verifyPasscode(passcode: string) {
-    return fetchToken(
-        'temp-name',
-        'temp-room',
-        passcode,
-        false /* create_room */
-    ).then(async (res) => {
-        const jsonResponse = await res.json()
-        if (res.status === 401) {
-            return { isValid: false, error: jsonResponse.error?.message }
-        }
-
-        if (res.ok && jsonResponse.token) {
-            return { isValid: true }
-        }
-    })
+    return Promise.resolve({ isValid: true, error: '' })
```

Comment:
we verify the room password when calling `join`, so for the moment I bypassed this one